src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -88,45 +146,99 @@ public void initializeConverter() {
+                float hue = color.getHue().floatValue() / 360.0f;
+                float saturation = color.getSaturation().floatValue() / 100.0f;
+                // float xy[] = ColorHelper.hsv2xy(hue, saturation, 1.0f);
+                float rgb[] = ColorHelper.hsv2rgb(hue, saturation, 1.0f);
+                float xy[] = ColorHelper.rgb2xy(rgb[0], rgb[1], rgb[2]);
+                logger.debug("{}: Calculated RGB. {}, {}, {}", endpoint.getEndpointAddress(), rgb[0], rgb[1], rgb[2]);
+                logger.debug("{}: Calculated XY. {}, {}", endpoint.getEndpointAddress(), xy[0], xy[1]);
+                int x = (int) (xy[0] * 65536.0f + 0.5f); // up to 65279
+                int y = (int) (xy[1] * 65536.0f + 0.5f); // up to 65279
+
+                clusterColorControl.moveToColorCommand(x, y, 10).get();
+            }
+
             @Override
             public void run() {
                 if (initialised == false) {
                     return;
                 }
 
-                int level;
                 try {
-                    if (command instanceof HSBType) {
-                        HSBType hsb = (HSBType) command;
-                        clusterColorControl
-                                .moveToHueAndSaturationCommand((int) (hsb.getHue().doubleValue() * 254.0 / 360.0 + 0.5),
-                                        (int) (hsb.getSaturation().doubleValue() * 254.0 / 100.0 + 0.5), 10)
-                                .get();
-                        level = hsb.getBrightness().intValue();
-                    } else if (command instanceof PercentType) {
-                        level = ((PercentType) command).intValue();
-                    } else if (command instanceof OnOffType) {
-                        level = (OnOffType) command == OnOffType.ON ? 100 : 0;
-                    } else {
-                        return;
+                    if (command instanceof OnOffType) {
+                        changeOnOff((OnOffType) command);
+                    }
+                    else if (command instanceof PercentType) {
+                        changeBrightness((PercentType) command);
                     }
+                    if (command instanceof HSBType) {
+                        HSBType color = (HSBType) command;
+                        PercentType level = color.getBrightness();
+
+                        if(level != currentHSB.getBrightness()) {
+                            changeBrightness(level);
+                            // Wait for transition to complete
+                            // Some lights do not like receiving a level/color change command
+                            // while the previous transition is in progress...
+                            Thread.sleep(1100);
+                        }
 
-                    clusterLevelControl.moveToLevelWithOnOffCommand((int) (level * 254.0 / 100.0 + 0.5), 10).get();
```

Comment:
Why did you change this? Doesn't it work with some bulbs? I would have preferred to reduce the number of dependant clusters being used unless it's necessary to also use the OnOff?

Reply:
It is still there, in the "changeBrightness" method. I had to change a bit the flowr (and add a delay if both the level and color are changed) as at least the IKEA lights do not like receiving a color update while still changing color (and the other way round). It is on the previous commentes in the code

                        if(level != currentHSB.getBrightness()) {
                            changeBrightness(level);
                            // Wait for transition to complete
                            // Some lights do not like receiving a level/color change command
                            // while the previous transition is in progress...
                            Thread.sleep(1100);
                        }

                        if(supportsHue) changeColorHueSaturation(color);
                        else changeColorXY(color);

A different question is why adding the onoff cluster. The reason is to be able to listen to onoff commands and update the channel accordingly (i.e. I may turn the light on just with the IKEA controller, and the binding would not realise).

But I can make this optional, so that if the cluster is not there, ignore it instead of failing.

Reply:
I thought I read in the ZCL docs that if a device supports both OnOff and LEvelControl, that they should be linked - so if you send an OnOff command to turn off the device, then I thought that LevelControl report would also be issued? Are you saying that this is not the case in the Ikea bulbs at least or have you not checked this?

If the Ikea bulbs don't work this way, then I'd be happy to leave the OnOff cluster in the converter, and just ignore this somehow if the OnOff isn't supported (I'm not sure how often OnOff won't be supported if LevelControl is, but it just seems best to reduce dependancies if possible)

Reply:
As far as I remember OnOff and LevelControl are linked, but sending an On/Off command does not need to report a change in Level, as the level does get changed necessarily by default when an On/Off command is sent (last level is used when On). At least the Ikea lights do not send a level attribute update here, and either my flspp led RGB controller.

So I think we will have to keep onoff cluster here